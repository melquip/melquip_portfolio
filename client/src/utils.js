export const loadState = (keys) => {
	try {
		let serializedState = null;
		let err = false;
		serializedState = keys.reduce((serialSt, key) => {
			serialSt[key] = localStorage.getItem('melquip_' + key);
			if(serialSt[key] !== null) {
				serialSt[key] = JSON.parse(serialSt[key]);
			} else {
				err = true;
			}
			return serialSt;
		}, {})
		return serializedState === null || err ? undefined : serializedState;
	} catch (err) {
		console.error('loadState:', err);
		return undefined;
	}
}

export const saveState = (state, keys) => {
	try {
		let serializedState = null;
		if (keys.length > 0) {
			keys.forEach(key => {
				serializedState = JSON.stringify(state[key]);
				localStorage.setItem('melquip_' + key, serializedState);
			});
		} else {
			serializedState = JSON.stringify(state);
			localStorage.setItem('melquip_state', serializedState);
		}
	} catch (err) {
		throw new Error('Not able to set local storage: ', err);
	}
}