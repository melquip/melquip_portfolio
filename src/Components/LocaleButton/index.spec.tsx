import React from "react";
import { render } from "@Test/utils";
import { LocaleButton } from "./index";

describe("Components", () => {
    describe("LocaleButton with isActive=true", () => {
        const { container } = render(
            <LocaleButton lang="tr" isActive onClick={() => {}} />
        );

        it("should match snapshot", () => {
            expect(container).toMatchSnapshot();
        });
    });

    describe("LocaleButton with isActive=false", () => {
        const { container } = render(
            <LocaleButton lang="tr" isActive={false} onClick={() => {}} />
        );

        it("should match snapshot", () => {
            expect(container).toMatchSnapshot();
        });
    });
});
