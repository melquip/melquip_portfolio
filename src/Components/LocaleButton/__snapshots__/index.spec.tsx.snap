// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components LocaleButton with isActive=false should match snapshot 1`] = `<div />`;

exports[`Components LocaleButton with isActive=true should match snapshot 1`] = `
.c1 {
  cursor: pointer;
  color: #2c3e50;
}

.c0 {
  color: #2c3e50;
}

<div>
  <div
    class="c0 c1"
  >
    tr
  </div>
</div>
`;
